CMakeLists.txt

```diff
@@ -376,6 +380,15 @@ if (MI_BUILD_OBJECT)
           RENAME ${mi_basename}${CMAKE_C_OUTPUT_EXTENSION} )
 endif()
 
+# pkg-config file support
+include("cmake/JoinPaths.cmake")
+join_paths(includedir_for_pc_file "\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}")
```

Comment:
I really don't like adding a file just for join_paths. We can just skip that here and always use "/" right?

Reply:
If `CMAKE_INSTALL_PREFIX="/usr"` and `CMAKE_INSTALL_INCLUDEDIR="/usr/include", then `${prefix}/usr/include` would expand to `/usr/usr/include`, which would be broken.

CMAKE_INSTALL_FULL_INCLUDEDIR does always contain the absolute path, but cmake doesn't have a mechanism for going the other way and guaranteeing that the path is relative to prefix. pkg-config works best (handling relocation etc.) when any containing prefix is replaced by `${prefix}`, so if possible we really want to do that...

The JoinPaths function makes sure that `${prefix}` is only added *iff* it is safe (because CMAKE_INSTALL_INCLUDEDIR is a relative path).

Reply:
The only two alternatives to JoinPaths are:
- to simply drop the use of `${prefix}` and write CMAKE_INSTALL_FULL_INCLUDEDIR directly to the pkg-config file, but like I said, that is non-idiomatic for pkg-config files.
- cmake 3.20 has added a builtin join_paths function, if you bump the minimum requirements.